server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -244,7 +269,7 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
                 public BulkScorer bulkScorer(LeafReaderContext context) throws IOException {
                     BulkScorer in = weight.bulkScorer(context);
                     if (in != null) {
-                        return new CancellableBulkScorer(in, checkCancelled);
+                        return new CancellableBulkScorer(in, () -> checkCancelled());
```

Comment:
Let's use a method ref instead in order to avoid creating an anonymous class
```suggestion
                        return new CancellableBulkScorer(in, cancellable::checkCancelled);
```